### ISSUE TITLE
Design: Events: Editors adding email instead of URL in "Call to Action", causing broken link alerts

### ISSUE THREAD

**Issue by stefaniefgray** [department-of-veterans-affairs/va.gov-cms#13329]
## Describe the defect
As an editor on https://prod.cms.va.gov/ creating a new Event, I would like attendees to RSVP by emailing an address such as VANJHCSCommsTeam [at] va.gov.

Unfortunately, if an email address is added as the "Call to Action" URL without the `mailto:` prefix, it saves as `/VANJHCSCommsTeam%40va.gov`

**Note:** The help text for adding an Event "Call to Action" URL does not advise editors to add `mailto:` before email addresses.

This issue is inconvenient for site visitors who may wish to click the link to RSVP.

This issue also creates broken link alerts that are reported every 30-60 minutes in #content-broken-links until they are resolved by the editor or by a member of the VA Drupal CMS Help Desk team.

Sample broken link alert created by this issue provided in screenshot below.

## To Reproduce
1. Create a new Event
2. From the "Call to Action" dropdown menu, select "Register"
3. Add an email address to the URL field
4. Click "Save"
5. Call to Action URL will save with improper formatting.

## Expected behavior
Email addresses stay true to original formatting when added as an Event "Call to Action" URL without the `mailto:` prefix.

## Screenshots
**Real life example:**
From https://prod.cms.va.gov/new-jersey-health-care/events/55340

<img width="568" alt="Screen Shot 2023-04-17 at 10 16 16 PM" src="https://user-images.githubusercontent.com/9059428/232670973-48b25798-63f1-4e8d-a065-f7588666cf26.png">

<img width="635" alt="Screen Shot 2023-04-17 at 10 25 02 PM" src="https://user-images.githubusercontent.com/9059428/232671091-3f3715fe-e6f2-49ee-a45c-0daccbfc0366.png">

From #content-broken-links:
Broken link report: https://dsva.slack.com/archives/C030F5WV2TF/p1681777264243279

<img width="635" alt="Screen Shot 2023-04-17 at 10 25 02 PM" src="https://user-images.githubusercontent.com/9059428/232671164-389bfa27-0079-4f27-847a-98701cae27d6.png">

 ## Additional context
If this issue is declined, the help text for Event "Call to Action" URLs should be updated to reflect that email addresses require the `mailto:` prefix.

### Suggested ACs from Help Desk
- [ ] Linkit instances enable option (in comments) to recognize email addresses and automatically add `mailto:`
- [ ] Add validation if possible
- [ ] (optional) Help text utilizes patch (in comments) to clarify this possibility to Editors

## Design

[Design file in Figma](https://www.figma.com/file/8ZT1nnu8r2twbOlEhysnNp/Events?type=design&node-id=133%3A575&mode=design&t=YRB5WXJtyy13xlBc-1)

## Design Acceptance Criteria
- [x] In Figma, Proposed a near-term fix across help text, validation and error messaging, and any other Drupal UI enhancements to either  1) prevent emails from being used at all or 2) ensure that the UI consistently delivers email addresses as a working Veteran experience.
- [ ] Met with PM and PO to solidify and ticket next steps

**Comment by swirtSJW** [comment]
Is there a reason this is facilities specific?   This seems like a PW thing.

**Comment by stefaniefgray** [comment]
@swirtSJW Only because it's an Event, otherwise please feel free to pass to PW @jilladams

**Comment by swirtSJW** [comment]
![image](https://user-images.githubusercontent.com/5752113/232822634-458ba144-44d1-47e2-91de-f897033a07b0.png)

Product ownership iis spec'ed out here. 
 I think it is our source of truth. https://app.mural.co/t/departmentofveteransaffairs9999/m/vagov6717/1653588059793/7a260edabca9535d25d71fac1035c979261c223c?sender=u18053dffba72153d0f939228

**Comment by kmariepat-cityfriends** [comment]
A note was made in scrum that this card may be for PW instead

**Comment by wesrowe** [comment]
@swirtSJW, has the cause been identified or was this the first time you heard of such an issue?

@stefaniefgray, this seems like it has probably been causing broken links for a while – is the behavior of adding an email as a CTA not very common?

**Comment by stefaniefgray** [comment]
@wesrowe This has definitely been an issue for a while and comes up every few weeks or so

**Comment by wesrowe** [comment]
@dsasser, I assigned to you for pre-refinement since it seems like drupal is doing the bad handling here.

**Comment by jilladams** [comment]
Paging @timcosgrove and/or @ndouglas from a CMS Collab POV as FYI / in case you have opinions here.

**Comment by ndouglas** [comment]
I can't think of any reason we'd want to treat some field instances differently, but I might have a blind spot in my reasoning 🙂  IMHO add the matcher to the default profile as you suggested, @dsasser.

**Comment by wesrowe** [comment]
Thanks for this analysis @dsasser. So right now, anywhere an Editor adds an email address as an "external link" in Linkit, across all these 20+ instances, they create a broken experience for a Veteran. Is that right? If so, I don't see how it would do any harm to make it a proper, not-broken mailto link. So my answer to your first decision point is: Let's apply it to every Linkit instance. 

I think improving the help text sounds like a good idea also. I'll update the ACs and we can review with Dave in a refinement or planning meeting.

**Comment by dsasser** [comment]
>  So right now, anywhere an Editor adds an email address as an "external link" in Linkit, across all these 20+ instances, they create a broken experience for a Veteran. Is that right?

I can't say this with 100% certainty because I haven't made connections from the linkit fields to the requisite frontend templates, but it is very likely. Fortunately it is easier to apply the change to all linkit fields globally than to create one-off profiles.

**Comment by jilladams** [comment]
From sprint 89 planning: 
* DaveC question: should we allow email addresses in this field? Probably not. Instead: disallow in this field, and provide an alternate mechanism for managing Registrations that are not an email address.
* Refinement things to do:  
    * Figure out with @stefaniefgray and @TroyA6 how we can assess how common this workflow is. 
    * Think about what the editor workaround is for managing non-URL registration flows. 
    * New paradigms may mean: CMS Collab Cycle

**Comment by davidconlon** [comment]
Hi folks - chiming in here with thoughts/opinions.
the "Expected behavior" seems a little 'thumb on the scale' potentially. 

> "Email addresses stay true to original formatting when added as an Event "Call to Action" URL without the mailto: prefix."

**An email address is not a URL.**

Instead, we should be thinking along the lines of
> Expected behavior: As an editor, I expect to be able to provide an email address to Veterans as a means to Register/RSVP, etc.

This then empowers the product team to make changes as appropriate. I can see a few things: 
1. We probably want to know when it's a URL vs an email address so that we can validate things differently
2. There is specific guidance we likely want to provide re: email addresses.
3. There may be different front-end experiences we want to provide to people depending on the different cases. For example the CTA shouldn't necessarily bury the email address but instead align to the correct VA.gov pattern.

**Comment by davidconlon** [comment]
FWIW this is not a defect, this is an enhancement request.

**Comment by wesrowe** [comment]
@stefaniefgray, we refined this with the UX team. A couple questions:

You said above this happens every couple weeks
1. What do you do when you see a broken link caused by an email address in CTA? 
2. If you're asking Editors to correct, do they typically know how they would request RSVPs (or whatever action) using a proper url?
3. Is the link corrected quickly, before it impacts Veterans?

cc @aklausmeier @thejordanwood @cindymerrill

**Comment by stefaniefgray** [comment]
Hi @wesrowe -- thanks for refining it!

Question 1: What do you do when you see a broken link caused by an email address in CTA?
- I go in and edit the node, and add the 'mailto:' prefix.

Question 2: If you're asking Editors to correct, do they typically know how they would request RSVPs (or whatever action) using a proper url?
- After manually correcting the link, we typically reach out to them via Jira Service Desk and let them know "Hey, you've got to add the 'mailto:' prefix before email addresses."

Question 3: Is the link corrected quickly, before it impacts Veterans?
- Yes, if it's coming up on #content-broken-links then our team is aware and can fix it post-haste

**Comment by wesrowe** [comment]
Thanks @stefaniefgray – one of the options for improving this situation is to _only_ allow URLs, per the intended design. I.e., add validation to reject email addresses. More questions!

1. Which CTA option(s) do Editors typically select when they use an email address as the URL? (register, apply, rsvp, more details)
2. Do you know what method you would recommend to an Editor for using the URL field for the use case(s)?

**Comment by wesrowe** [comment]
@stefaniefgray, just pinging you again on this. 

@thejordanwood, is there anything else you need to know before starting on this?

cc @aklausmeier

**Comment by thejordanwood** [comment]
@wesrowe Nothing else from me. I'm ready to start this!

**Comment by stefaniefgray** [comment]
Wow, I'm sorry @wesrowe -- I typed this and never clicked "Comment" 🤦‍♀️ 

1. _"Which CTA option(s) do Editors typically select when they use an email address as the URL? (register, apply, rsvp, more details)"_
Typically, Register or RSVP (but honestly, e-mail addresses ought to work for all selections, just in case)

2. _Do you know what method you would recommend to an Editor for using the URL field for the use case(s)?_
At this time, I'd just recommend the "mailto:" workaround. It might help to maybe break it out into two separate fields, one where an email address could be used and one where a regular URL could be used?

**Comment by thejordanwood** [comment]
@wesrowe The design is ready for your review. The mockups can be viewed in the Call to Action tab of [this Figma file](https://www.figma.com/file/8ZT1nnu8r2twbOlEhysnNp/Events?type=design&node-id=133%3A575&mode=design&t=vjlYiATUIhNOrFsz-1). 

@aklausmeier and @laflannery have approved this design. 

I have a couple of notes on this and what's changing:

1. New experience: After the Call to Action dropdown, a new dropdown will appear that asks editors if they want to use a URL or an email address. The one they select determines which field appears after that. This will be the same experience if they select Register, Apply, RSVP, or More Details.
2. From Laura: We will want to make sure the focus follows the appropriate tab order when the new field is displayed.
3. Help text changes: This section was called “Register”, but I changed the name to “How to join” to make it more general, since the call to action dropdown also includes Apply, RSVP, or More Details. I changed some field labels and help text to reflect this.
4. I think we may need to change the wording on the "Registration is required for this event" checkbox to make it more general as well. However, I'm having a hard time actually seeing what this checkbox does. I don't notice any changes on the FE and can't tell if it's specifically tied to the Register CTA or all of the CTAs. Not sure if I'm just missing something here, or if an engineer would need to look into this more.

**Comment by wesrowe** [comment]
A gap in our understanding that need FE dev input:

We need to know how/if the "registration required" checkbox in the Event UI is used by the FE template.

Example:
- https://www.va.gov/outreach-and-events/events/61560/
- https://www.va.gov/outreach-and-events/events/61560/

The Event Editor checked the box for required, but I don't see anything in the FE that indicates that:
<img width="812" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/4054752/fe53eca7-d6a6-4759-82cc-e0d999277b3e">
<img width="375" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/4054752/d589c36a-4389-4f53-9b27-81b02fd3f35c">

@chriskim2311 or @randimays, is it a pre-refinement (low-lift) activity to see whether the field from Drupal is being used in the template? If so, please check it out and comment here. Thanks!

**Comment by thejordanwood** [comment]
Closing this ticket. [A new ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15388) has been made to find our if the "registration required" checkbox is being used anywhere.

**Comment by wesrowe** [comment]
Aaaand note: the question I asked FE engineers in the comment above is one of the ACs in the new ticket.
